Silicon/NXP/iMX6Pkg/Drivers/GOP/CPMem.h

```diff
@@ -0,0 +1,276 @@
+    CPMEM_BM_BW_BH_8,
+    CPMEM_BM_BW_BH_16,
+    CPMEM_BM_UNUSED,
+} CPMEM_BM;
+
+typedef enum {
+    CPMEM_ROT_NO_ROTATION,
+    CPMEM_ROT_90_ROTATION,
+} CPMEM_ROT;
+
+typedef enum {
+    CPMEM_HF_NO_HFLIP,
+    CPMEM_HF_HFLIP_ENABLE,
+} CPMEM_HF;
+
+typedef enum {
+    CPMEM_HF_NO_VFLIP,
+    CPMEM_HF_VFLIP_ENABLE,
+} CPMEM_VF;
+
+typedef enum {
+    CPMEM_THE_DISABLE,
+    CPMEM_THE_ENABLE,
+} CPMEM_THE;
+
+typedef enum {
+    CPMEM_CAP_SKIP_LOW,
+    CPMEM_CAP_SKIP_HIGH,
+} CPMEM_CAP;
+
+typedef enum {
+    CPMEM_CAE_DISABLE,
+    CPMEM_CAE_ENABLE,
+} CPMEM_CAE;
+
+typedef enum {
+    CPMEM_EOLI_DISABLE,
+    CPMEM_EOLI_ENABLE,
+} CPMEM_EOLI;
+
+typedef enum {
+    CPMEM_PFS_NON_INT_444,
+    CPMEM_PFS_NON_INT_422,
+    CPMEM_PFS_NON_INT_420,
+    CPMEM_PFS_PAR_INT_422,
+    CPMEM_PFS_PAR_INT_420,
+} CPMEM_PFS_PLANAR;
+
+typedef enum {
+    CPMEM_DEC_SEL_0_15,
+    CPMEM_DEC_SEL_64_79,
+    CPMEM_DEC_SEL_128_143,
+    CPMEM_DEC_SEL_192_207,
+} CPMEM_DEC_SEL;
+
+typedef enum {
+    CPMEM_PFS_CODE = 5,
+    CPMEM_PFS_GENERIC_DATA,
+    CPMEM_PFS_RGB,
+    CPMEM_PFS_INT_422_Y1U1Y2V1,
+    CPMEM_PFS_INT_422_Y2U1Y1V1,
+    CPMEM_PFS_INT_422_U1Y1V1Y2,
+    CPMEM_PFS_INT_422_U1Y2V1Y1,
+} CPMEM_PFS_PACKED;
+
+typedef enum {
+    CPMEM_ALU_SAME_BUFFER,
+    CPMEM_ALU_SEPERATE_BUFFER,
+} CPMEM_ALU;
+
+typedef enum {
+    CPMEM_ID_ID_0,
+    CPMEM_ID_ID_1,
+    CPMEM_ID_ID_2,
+    CPMEM_ID_ID_3,
+} CPMEM_ID;
+
+typedef enum {
+    CPMEM_CRE_DISABLE,
+    CPMEM_CRE_ENABLE,
+} CPMEM_CRE;
+
+#pragma pack(push, 1)
+
+// CPMEM_WORD0_PLANAR - Non interlaved
+typedef union {
+    struct {
+        UINT32 XVVirtualCoordinate : 10;
+        UINT32 YVVirtualCoordinate : 9;
+        UINT32 XBinnerBlockCoordinate : 13;
+        UINT32 YBinnerBlockCoordinate : 12;
+        UINT32 NewSubBlock : 1;
+        UINT32 CurrentField : 1;
+        UINT32 MemUBufferOffset : 22;
+        UINT32 MemVBufferOffset : 22;
+        UINT32 InitialOffsetX : 4;
+        UINT32 ReduceDoubleReadOrWrites : 1;
+        UINT32 Reserved1 : 18;
+        UINT32 ScanOrder : 1;
+        UINT32 BandMode : 3;
+        UINT32 BlockMode : 2;
+        UINT32 Rotation : 1;
+        UINT32 HorizontalFlip : 1;
+        UINT32 VerticalFlip : 1;
+        UINT32 ThresholdEnable : 1;
+        UINT32 ConditionalAccessPolarity : 1;
+        UINT32 ConditionalAccessEnable : 1;
+        UINT32 FrameWidth : 13; // Max 8192 Pixel
+        UINT32 FrameHeight : 12; // Max 4096 Pixel
+        UINT32 EndOfLineInterrupt : 1;
+        UINT32 Reserved2 : 9;
+    };
+    UINT32 Word0[5];
+} CPMEM_WORD0_PLANAR_REG;
+
+// CPMEM_WORD1_PLANAR - Non interleaved
+typedef union {
+    struct {
+        UINT32 ExtMemBuffer0Address : 29;
+        UINT32 ExtMemBuffer1Address : 29;
+        UINT32 InterlaceOffset : 20;
+        UINT32 NumberOfPixelsInWholeBurstAccess : 7;
+        UINT32 PixelFormatSelect : 4;
+        UINT32 AlphaUsed : 1;
+        UINT32 AlphaChannelMapping : 3; // Doc error? Thisis documented as 2 bit wide

```

Comment:
Delete this comment. The bitfield breakdown agrees with the 3-bit wide field and it matches with the adjacent fields too, the size is just labeled incorrectly.

Reply:
fixed